cumulusci/utils/clean_invalid_references.py

```diff
@@ -0,0 +1,365 @@
+
+    return fields, recordTypes
+
+
+def get_tabs_from_app(root: ET.Element):
+    tabs = set()
+    for tab in root.findall("tabs"):
+        tabs.update([tab.text])
+    return tabs
+
+
+def get_target_entities_from_zip(zip_src: zipfile.ZipFile):
+    target_entities = {key: set() for key in FOLDER_PERM_DICT.keys()}
+    for name in zip_src.namelist():
+        if name == "package.xml":
+            continue
+        metadataType = name.split("/")[0]
+        metadataName = name.split("/")[1].split(".")[0]
+        target_entities[metadataType].update([metadataName])
+
+        # If object, fetch all the fields and record types present inside the file
+        if name.endswith(".object"):
+            file = zip_src.open(name)
+            root = ET.parse(file).getroot()
+            root = strip_namespace(root)
+            fields, recordTypes = get_fields_and_recordtypes(root, metadataName)
+            target_entities["fields"].update(fields)
+            target_entities["recordTypes"].update(recordTypes)
+
+        # To handle tabs which are not part of TabDefinition Table
+        if name.endswith(".app"):
+            file = zip_src.open(name)
+            root = ET.parse(file).getroot()
+            root = strip_namespace(root)
+            target_entities["tabs"].update(get_tabs_from_app(root))
+
+    return target_entities
+
+
+def clean_zip_file(zip_src: zipfile.ZipFile, target_entities: Dict[str, set]):
+    zip_dest = zipfile.ZipFile(io.BytesIO(), "w", zipfile.ZIP_DEFLATED)
+    for name in zip_src.namelist():
+        file = zip_src.open(name)
+        if any(
+            item.extension in name and name.startswith(item.folder_name)
+            for item in FILES_TO_BE_CLEANED
+        ):
+            root = ET.parse(file).getroot()
+            cleaned_root = CleanXML(root, target_entities)
+            tree = ET.ElementTree(cleaned_root)
+            cleaned_content = io.BytesIO()
+            tree.write(
+                cleaned_content,
+                pretty_print=True,
+                xml_declaration=True,
+                encoding="utf-8",
+            )
+            zip_dest.writestr(name, cleaned_content.getvalue())
+        else:
+            zip_dest.writestr(name, file.read())
+
+    return zip_dest
+
+
+def strip_namespace(element: ET.Element):
+    for elem in element.iter():
+        if "}" in elem.tag:
+            elem.tag = elem.tag.split("}", 1)[1]
+    return element
+
+
+def fetch_permissionable_entity_names(
+    root: ET.Element, perm_entity: PermissionElementXPath, parent: bool = False
+):
+    entity_names = set()
+    for element in root.findall(perm_entity.permission_xpath):
+        entity_names.add(
+            perm_entity.return_parent(element)
+            if parent
+            else perm_entity.return_name(element)
+        )
+    return entity_names
+
+
+def CleanXML(root: ET.Element, target_entities: Dict[str, set]):
```

Comment:
```suggestion
def clean_xml(root: ET.Element, target_entities: Dict[str, set]):
```

I'd use this form to match typical Python idiom. `CleanXML` looks like a class name.

Reply:
Yep will change that